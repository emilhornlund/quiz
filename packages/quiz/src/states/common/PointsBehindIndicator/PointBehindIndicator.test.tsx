import { render, screen, within } from '@testing-library/react'
import React from 'react'
import { describe, expect, it } from 'vitest'

import PointBehindIndicator from './PointsBehindIndicator'

describe('PointBehindIndicator', () => {
  it('renders points and nickname with strong emphasis', () => {
    render(<PointBehindIndicator points={12} nickname="Alice" />)

    const container = screen.getByText(/points behind/i).closest('div')!
    expect(container).toBeInTheDocument()

    const strongs = within(container).getAllByRole('strong', { hidden: true })
    expect(strongs).toHaveLength(2)
    expect(strongs[0]).toHaveTextContent('12')
    expect(strongs[1]).toHaveTextContent('Alice')

    expect(container).toMatchSnapshot()
  })

  it('renders another set of props and matches snapshot', () => {
    const { container } = render(
      <PointBehindIndicator points={1} nickname="Bob the Builder" />,
    )

    expect(screen.getByText(/points behind/i)).toHaveTextContent(
      '1 points behind Bob the Builder',
    )

    expect(container).toMatchSnapshot()
  })
})
