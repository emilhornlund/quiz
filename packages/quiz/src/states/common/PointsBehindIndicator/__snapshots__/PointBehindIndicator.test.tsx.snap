// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PointBehindIndicator > renders another set of props and matches snapshot 1`] = `
<div>
  <div
    class="pointsBehindIndicator"
  >
    <strong>
      1
    </strong>
     points behind 
    <strong>
      Bob the Builder
    </strong>
  </div>
</div>
`;

exports[`PointBehindIndicator > renders points and nickname with strong emphasis 1`] = `
<div
  class="pointsBehindIndicator"
>
  <strong>
    12
  </strong>
   points behind 
  <strong>
    Alice
  </strong>
</div>
`;
